robot/impl/local_robot.go

```diff
@@ -1183,6 +1184,37 @@ func (r *localRobot) reconfigure(ctx context.Context, newConfig *config.Config,
 
 	var allErrs error
 
+	// Maintenance config can be configured to block reconfigure based off of a sensor reading
```

Comment:
Thanks for encoding rules here. Some questions:
- Where is the logic for the priority described below?
- Will this documentation be exposed to users in any other way besides this comment?

Reply:
1. The logic is spread out a bit. Some is controlled by how we start up resources when there are conflicts. The other parts of the logic depends on how sensors are returned from ResourceFromRobot when there are conflicts
2. This is a good point. I think we should expose this logic to users. I think the frontend component would be a good place to share this info

Reply:
I buy that they all just "work" without much changes. How does this one happen?

> Main robot and fragment share sensor name -> main robot sensor is chosen

I wasn't aware that we had any special logic around fragments.

> This is a good point. I think we should expose this logic to users. I think the frontend component would be a good place to share this info

SGTM; as long as that's captured in some eventual fleet ticket.

Reply:
ahh forgot to mention the fragments. The logic for that is in the frontend when we apply a fragment to a config. so I guess the logic is spread out even more 😅 

Also 👍 will add it to the front end ticket